### PR TITLE
Add cancellable upload API, onProgress and filename options

### PR DESCRIPTION
**Pull request by ggurgone**
This PR adds 

* the ability to cancel uploads for `createUploadPath`, `uploadFile` and `uploadImage`
* track the upload progress with an ad-hoc `onProgress` callback 
* option to assign (rename) a custom name to the uploaded file

### CODE REVIEW

File: test/src/uploadFile_test.js [hunk 1]
```diff
@@ -95,4 +95,79 @@ describe('Upload file from', async () => {
...
+  context('remote upload cancellation', () => {
+    it(
+      'can be cancelled',
+      vcr(async () => {
+        let noProgress = true;
+        const promise = uploadFile(
+          client,
+          'https://www.datocms-assets.com/13095/1561723946-happyfoxbymazack-d8u2l0s-2.jpeg',
+          {},
+          {},
```
Comment (ggurgone) [inline test/src/uploadFile_test.js:108]: To preserve backwards compatibility we need to pass these guys when we want to pass `onProgress`. This is unfortunate but if you think it makes sense I can explore the option to overload it i.e. make it work with the old signature and a new one:

```js
uploadFile(client, string | File, options)
```

With this new signature `options` will include the other options as well

File: src/upload/adapters/nodeRemote.js [hunk 2]
```diff
@@ -6,21 +6,77 @@ import url from 'url';
...
+                payload: { percent },
+              });
+            },
       })
-        .then(response => {
+        .then(async response => {
+          let data;
+          // Axios' onDownloadProgress works only in the browser
+          // so in Node.js we need to implement it ourselves with streams.
+          if (options.onProgress) {
```
Comment (ggurgone) [inline src/upload/adapters/nodeRemote.js:55]: let me know if you can think of a cleaner way to do this

File: test/src/uploadFile_test.js (hunk 1 above)
Comment (stefanoverna) [reply test/src/uploadFile_test.js:108]: Never really liked the `uploadFile` method in the first place, as it tries to do too many things togheter (create upload request, upload to S3, create the upload, return a Single asset file compatible object).. I'm planning to remove it from the docs in favour of the more explicit `.createUploadPath` and `.uploads.create` methods, so don't worry about it! 👍

File: src/upload/adapters/nodeRemote.js (hunk 2 above)
Comment (stefanoverna) [reply src/upload/adapters/nodeRemote.js:55]: not really an expert on streams here :/

**Comment by ggurgone** [comment]
@stefanoverna please take one last look and if everything looks good approve this PR so that I can merge it into `feature/enhance-upload-api`

**Review by stefanoverna — APPROVED** [review]
Looks good to me @ggurgone! Thanks!

**Pull request merged**